app/models/batch.rb

```diff
@@ -31,6 +31,10 @@
 class Batch < ApplicationRecord
   before_create :create_uuid
 
+  def self.last_scored(network)
+    where('network = ? and scored_at IS NOT NULL', network).last
```

Comment:
Don't you need some ordering here or last is enough?

Reply:
Default is the order by id, so if I understand it correctly it should always point to the newly created object.